File: Selenium/Project/CRM_Activity8.java
package Project;

import java.time.Duration;
import java.util.ArrayList;
import java.util.List;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.Assert;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;

//Traversing tables
//Goal: Open the accounts page and print the contents of the table

public class CRM_Activity8 {
	WebDriver driver;
	WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));

	// Set up method : Open a browser
	@BeforeClass
	public void setUp() {
		// Initialize Chrome driver
		driver = new ChromeDriver();
		// Navigate to ‘http://alchemy.hguy.co/crm’.
		driver.get("http://alchemy.hguy.co/crm");
	}

	@Test(priority = 1)
	public void login() {
		driver.findElement(By.id("user_name")).sendKeys("admin");
		driver.findElement(By.id("username_password")).sendKeys("pa$$w0rd");
		driver.findElement(By.id("bigbutton")).click();
		String homePageUrl = driver.getCurrentUrl();
		Assert.assertTrue(homePageUrl.contains("Home"), "Login Failed !");

	}

	@Test(priority = 2)
	public void openAccountsPage() throws InterruptedException {
		driver.manage().window().maximize();
//		Navigate to the Sales -> Accounts page
		driver.findElement(By.id("grouptab_0")).click();
		driver.findElement(By.id("moduleTab_9_Accounts")).click();
		Thread.sleep(2000);
		String actualAccountsPageHeading = driver.findElement(By.xpath("//h2[@class='module-title-text']")).getText()
				.strip();
		String expectedAccountsPageHeading = "ACCOUNTS";
		Assert.assertEquals(actualAccountsPageHeading, expectedAccountsPageHeading);
	}

	@Test(priority = 3)
	public void getAccountNames() throws InterruptedException {

		// create an empty list and add expected Account Names
		List<String> expectedAccountNames = new ArrayList<String>();
		expectedAccountNames.add("5D Investments");
		expectedAccountNames.add("A.D. Importing Company Inc");
		expectedAccountNames.add("AtoZ Co Ltd");
		expectedAccountNames.add("B.H. Edwards Inc");
		expectedAccountNames.add("Calm Sailing Inc");
		// System.out.println("list1:" + expectedAccountNames);

//		Find the table on the page and print the names of the first 5 odd-numbered rows of the table to the console
		List<WebElement> namesList = driver
				.findElements(By.xpath("//table[@class='list view table-responsive']/tbody/tr/td[3]"));
		List<String> actualAccountNames = new ArrayList<String>();
		// Start from index 1 and increment by 2 until index 9
		for (int i = 1; i < 10; i += 2) {
			actualAccountNames.add(namesList.get(i).getText());
		}
		// System.out.print("list2:" + actualAccountNames);
		Assert.assertEquals(actualAccountNames, expectedAccountNames);
	}

	@AfterClass
	public void tearDown() {
		// Close the browser.
		driver.close();
	}

}

File: Selenium/Project/CRM_Activity8.py
# Traversing tables
# Goal: Open the accounts page and print the contents of the table.


# Import webdriver from selenium
from selenium import webdriver
from selenium.webdriver.common.by import By
import time

# Start the Driver
with webdriver.Chrome() as driver:
  
    driver.get("https://alchemy.hguy.co/crm/")

# Find and select the username and password fields
# Enter login credentials into the respective fields
    driver.find_element(By.ID,"user_name").send_keys("admin")
    driver.find_element(By.ID,"username_password").send_keys("pa$$w0rd")
    driver.find_element(By.ID,"bigbutton").click()
    homePageHeader = driver.find_element(By.ID,"tab0").text
    # print(homePageHeader)
    if(homePageHeader=="SUITECRM DASHBOARD"):
        print("Home Page Opened Successfully !")
    else:
        print("Login failed !")

# Navigate to the Sales -> Accounts page
    driver.fullscreen_window()
    driver.find_element(By.ID,"grouptab_0").click()
    driver.find_element(By.ID,"moduleTab_9_Accounts").click()
    time.sleep(2)
# Find the table on the page and print the names of the first 5 odd-numbered rows of the table to the console
    print("The first 5 Odd-numbered names are : ")
    namesList = driver.find_elements(By.XPATH,"//table[@class='list view table-responsive']/tbody/tr/td[3]")
    for x in range(1, 10, 2):
        print(x,namesList[x].text)

#Close the browser
    driver.quit()
    





    